android/src/main/java/com/bitmovin/player/reactnative/offline/OfflineManager.kt

```diff
@@ -0,0 +1,208 @@
+package com.bitmovin.player.reactnative.offline
+
+import com.bitmovin.player.api.deficiency.ErrorEvent
+import com.bitmovin.player.api.offline.OfflineContentManager
+import com.bitmovin.player.api.offline.OfflineContentManagerListener
+import com.bitmovin.player.api.offline.options.OfflineContentOptions
+import com.bitmovin.player.api.offline.options.OfflineOptionEntry
+import com.bitmovin.player.api.offline.options.OfflineOptionEntryAction
+import com.bitmovin.player.api.offline.options.OfflineOptionEntryState
+import com.bitmovin.player.api.source.SourceConfig
+import com.bitmovin.player.reactnative.NativeId
+import com.bitmovin.player.reactnative.converter.JsonConverter
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.modules.core.DeviceEventManagerModule
+
+class OfflineManager(
+    private val nativeId: NativeId,
+    private val context: ReactApplicationContext,
+    private val offlineId: String,
+    source: SourceConfig,
+    location: String
+) :
+    OfflineContentManagerListener {
+
+    val contentManager: OfflineContentManager
+    var contentOptions: OfflineContentOptions? = null
+        private set
+
+    init {
+        contentManager = OfflineContentManager.getOfflineContentManager(
+            source, location, offlineId, this, context
+        )
+    }
+
+    fun getOptions() {
+        contentManager.getOptions()
+    }
+
+    /**
+     * Process the `OfflineDownloadRequest`.
+     * The `OfflineContentOptions` are stored in memory in this class because they can not be constructed to pass to the process call.
+     * We're copying the iOS interface for the `minimumBitrate` so that react-native will have a consistent interface.
+     *
+     */
+    fun process(request: OfflineDownloadRequest) {
+        if (contentOptions != null) {
+            contentOptions!!.videoOptions
+                .firstOrNull { option -> option.bitrate >= request.minimumBitrate }
```

Comment:
Not sure what is expected on iOS. The highest video quality, or the one closest to the `minimumBitrate`?

Reply:
That's actually a question for @rolandkakonyi 🙂

Reply:
The next highest from the minimum. E.g. of there is a ladder of `[32k, 64k, 120k]` and minimum is set to `50k` then it should pick `64k` as that is the next one after the minimum requirement.
Disclaimer: this is completely handled by Apple on iOS, so not sure if some other factors are considered or not.

Also not sure what happens if you set minimum to `150k`, if it would select the highest or fail the download.

Reply:
Based on a super-quick test, it seems to select the highest quality in case of overshooting.

Note that `videoOptions` should be ordered for this `firstOrNull` to succeed.

Reply:
Thanks for the input.

Yes, the code should be changed to first sort by the bitrate and then find the first above the `minimumBitrate` or if there is none, the one with the highest bitrate should be chosen.